modules/data.atmosphere/R/solar_angle.R

```diff
@@ -2,6 +2,50 @@
 #'
 #' For explanations of formulae, see https://web.archive.org/web/20180307133425/http://www.itacanet.org/the-sun-as-a-source-of-energy/part-3-calculating-solar-angles/
 #'
+#' @description
+#' `cos_solar_zenith_angle` returns the cosine of solar zenith angle.
+#' The cosine of the solar zenith angle (cossza) is the angle between the sun’s rays and the vertical.
+#' This angle is important because it determines the intensity of solar radiation that reaches the Earth's surface.
+#' When the solar zenith angle is small (i.e. the sun is close to directly overhead), the intensity of solar radiation is high.
+#' When the solar zenith angle is larger (i.e. the sun is closer to the horizon), the intensity of solar radiation is lower, resulting in cooler temperatures.
+#' It is required for photolysis.
+#'
+#' It depends upon the following -
+#'
+#' doy Day of year : It is the day when you are calculating the angle.
+#' January 1st being day 1 and December 31st being day 365 (or 366 in leap years).
+#' It is used to determine the position of the sun in the sky throughout the year.
+#' Unit - Dimensionless (a simple count of the day within a year).
+#'
+#' hr Hours timestep : It is the time of day that you're trying to predict zenith angle for.
+#' It's an integer value ranging from 0 to 24.
+#' hour time step might be used to represent the change in solar position and angle between consecutive hours within a day.
+#' Unit - Hours (h)
+#'
+#' dt Timestep : It represents the time interval between two values.
+#' Therefore it is an integer value ranging from 0 to 60.
```

Comment:
1. doesn't have to be an integer
2. I don't think you need to specify the range
3. the "between two values" comment doesn't make sense if you are feeding the functions scalar values, which is what a lot of people would assume you do. I think it would be better to add something like "Used if you are passing the function a vector times and want to calculate radiation at the midpoint between timesteps"